src/qililab/platform/platform.py

```diff
@@ -574,13 +574,34 @@ def execute_qprogram(  # pylint: disable=too-many-locals
         calibration: Calibration | None = None,
         debug: bool = False,
     ) -> QProgramResults:
-        """Execute a QProgram using the platform instruments.
+        """Execute a :class:`.QProgram` using the platform instruments.
+
+        |
+
+        **The execution is done in the following steps:**
+
+        1. Compile the QProgram.
+        2. Run the compiled QProgram.
+        3. Acquire the results.
+
+        |
+
+        **The execution can be done for (buses associated to) two different type of clusters:**
+
+        - For ``Qblox`` modules, the compilation is done using the :class:`.QbloxCompiler`. Which compiles the :class:`.QProgram` into``Q1ASM`` for multiple sequencers based on each bus, uploads and executes the sequences, acquires the results and resets the settings by desynching each bus.
+        - For ``Quantum Machines`` clusters, the compilation is done using the :class:`.QuantumMachinesCompiler`. Which transforms the :class:`.QProgram` into ``QUA``, the compilation language for ``Quantum Machines``, runs the job created by the ``QUA`` execution and returns the results ordering them by bus.
```

Comment:
Grammar error, `which` cannot start a sentence.

```suggestion
        - For ``Quantum Machines`` clusters, the compilation is done using the :class:`.QuantumMachinesCompiler`. This compiler transforms the :class:`.QProgram` into ``QUA``, the programming language of ``Quantum Machines`` hardware. It then executes the resulting QUA program and returns the results, organized by bus.
```